core/trino-main/src/main/java/io/trino/connector/DynamicCatalogManagerModule.java

```diff
@@ -33,12 +36,16 @@ protected void setup(Binder binder)
         if (buildConfigObject(ServerConfig.class).isCoordinator()) {
             binder.bind(CoordinatorDynamicCatalogManager.class).in(Scopes.SINGLETON);
             CatalogStoreConfig config = buildConfigObject(CatalogStoreConfig.class);
-            switch (config.getCatalogStoreKind()) {
-                case MEMORY -> binder.bind(CatalogStore.class).to(InMemoryCatalogStore.class).in(Scopes.SINGLETON);
-                case FILE -> {
+            switch (config.getCatalogStoreKind().toLowerCase(Locale.ROOT)) {
+                case "memory" -> binder.bind(CatalogStore.class).to(InMemoryCatalogStore.class).in(Scopes.SINGLETON);
+                case "file" -> {
                     configBinder(binder).bindConfig(FileCatalogStoreConfig.class);
                     binder.bind(CatalogStore.class).to(FileCatalogStore.class).in(Scopes.SINGLETON);
                 }
+                default -> {
+                    newOptionalBinder(binder, CatalogStoreManager.class).setBinding().to(CatalogStoreManager.class).in(Scopes.SINGLETON);
+                    binder.bind(CatalogStore.class).to(CatalogStoreManager.class).in(Scopes.SINGLETON);
```

Comment:
I think you also need
```java
binder.to(CatalogStoreManager.class).in(Scopes.SINGLETON);
```
Otherwise you get different instances for `Optional<CatalogStoreManager>` and `CatalogStore` (or maybe I've always misunderstood Guice)

Reply:
From JavaDocs of `OptionalBinder` (https://google.github.io/guice/api-docs/4.2/javadoc/com/google/inject/multibindings/OptionalBinder.html) (emphasis mine).

> **_If neither setDefault nor setBinding are called, it will try to link to a user-supplied binding of the same type._** If no binding exists, the optionals will be absent. Otherwise, if a user-supplied binding of that type exists, or if setBinding or setDefault are called, the optionals will return present if they are bound to a non-null value.